### PR TITLE
Adding GitHub Issue/PR templates

### PR DESCRIPTION
**Pull request by jtaubensee**
Also updating contribution guidelines.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- A description of the changes the pull request makes is included, and a reference to the bug/issue the pull request fixes is included, if applicable
+- All files have the Microsoft copyright header
+
+#### Testing guidelines
+
+The following guidelines must be followed in **EVERY** pull request that is opened.
+
+- Pull request includes test coverage for the included changes
+- Tests must use xunit
+- Test code should not contain hard coded values for resource names, or similar values
```
Comment (vinaysurya) [inline .github/CONTRIBUTING.md]: Remove comma?

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## Actual Behavior
```
Comment (SeanFeldman) [inline .github/ISSUE_TEMPLATE.md]: Should "Expected Behavior" come before the "Actual Behavior"?
Usually when you run into an issue, you expect X, but observe Y 😄

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+If applicable, reference the bug/issue that this pull request fixes here.
+-->
+
+This checklist is used to make sure that common guidelines for a pull request are followed.
+
+- [ ] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
+- [ ] Title of the pull request is clear and informative.
+- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
+- [ ] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
```
Comment (SeanFeldman) [inline .github/PULL_REQUEST_TEMPLATE.md]: How one would verify it? Did you mean the tests are passing?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+- [ ] If applicable, the public code is properly documented.
```
Comment (SeanFeldman) [inline .github/PULL_REQUEST_TEMPLATE.md]: > If applicable, the public code is properly documented.

For public changes, code should be documented (xml doc).

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+- [ ] Pull request includes test coverage for the included changes.
```
Comment (SeanFeldman) [inline .github/PULL_REQUEST_TEMPLATE.md]: 👍

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+- [ ] The code builds without any errors.
```
Comment (SeanFeldman) [inline .github/PULL_REQUEST_TEMPLATE.md]: Should this be replaced by "all tests are green"? If they are, the code will compile w/o errors _and_ tests will pass.

**Pull request merged**